<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>762</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QGridLayout" name="readout_grid">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item row="2" column="0">
       <widget class="QLineEdit" name="readout_connect_line">
        <property name="placeholderText">
         <string>Target GPIB Adress</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="readout_voltage_line">
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="placeholderText">
         <string>Not Connected</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QLabel" name="readout_label">
        <property name="font">
         <font>
          <pointsize>20</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Intensity Readout</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="readout_voltage_label">
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Current Value:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="readout_connect_pushbutton">
        <property name="text">
         <string>Connect</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="MplWidget" name="readout_plot" native="true">
        <property name="minimumSize">
         <size>
          <width>640</width>
          <height>480</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QLabel" name="laser_label">
        <property name="font">
         <font>
          <pointsize>20</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Recording</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="2">
       <layout class="QHBoxLayout" name="recording_layout">
        <item>
         <widget class="QTextEdit" name="recordin_status_text">
          <property name="toolTip">
           <string notr="true"/>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="record_target_line">
          <property name="placeholderText">
           <string>Filename</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="toggle_recording_pushbutton">
          <property name="text">
           <string>Toggle Recording</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="tag_text_line">
          <property name="placeholderText">
           <string>Tag Text</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="tag_event_pushbutton">
          <property name="text">
           <string>Tag Event</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QGridLayout" name="signal_grid">
      <property name="horizontalSpacing">
       <number>0</number>
      </property>
      <item row="14" column="0">
       <widget class="QPushButton" name="dump_signal_setting_pushbutton">
        <property name="text">
         <string>Dump device Settings</string>
        </property>
       </widget>
      </item>
      <item row="14" column="1">
       <widget class="QPushButton" name="dump_err_pushbutton">
        <property name="text">
         <string>Dump Device Errors</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QLineEdit" name="to_frequency_line">
        <property name="text">
         <string/>
        </property>
        <property name="placeholderText">
         <string>To</string>
        </property>
       </widget>
      </item>
      <item row="20" column="0" colspan="4">
       <widget class="QTextEdit" name="output_textedit"/>
      </item>
      <item row="19" column="0" colspan="4">
       <widget class="QLabel" name="output_label">
        <property name="font">
         <font>
          <pointsize>20</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Output Console</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="9" column="1">
       <layout class="QHBoxLayout" name="signal_frequency_layout">
        <item>
         <widget class="QLineEdit" name="signal_frequency_line">
          <property name="placeholderText">
           <string>Frequency</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="frequency_units_combobox">
          <property name="placeholderText">
           <string>Units</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="15" column="0" colspan="4">
       <widget class="QLabel" name="external_modulation_label">
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>External Pulse Modulation Settings</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="11" column="0">
       <widget class="QComboBox" name="pulse_function_combobox">
        <property name="placeholderText">
         <string>Pulse Function</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLineEdit" name="from_frequency_line">
        <property name="placeholderText">
         <string notr="true">From</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="4">
       <widget class="QLabel" name="frequency_sampler_label">
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Frequency Sampler</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="3">
       <widget class="QLineEdit" name="target_ip_line">
        <property name="toolTip">
         <string notr="true" comment="Enter a target adress recognized by the device driver"/>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="placeholderText">
         <string>Target Adress</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0" colspan="3">
       <widget class="QProgressBar" name="sampler_progress">
        <property name="value">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="4">
       <widget class="QLabel" name="signal_generator_label">
        <property name="font">
         <font>
          <pointsize>20</pointsize>
          <underline>false</underline>
          <kerning>true</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">Signal Generator Settings</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="6" column="2">
       <widget class="QLineEdit" name="stepsize_frequency_line">
        <property name="placeholderText">
         <string>Step Size</string>
        </property>
       </widget>
      </item>
      <item row="10" column="0" colspan="4">
       <widget class="QLabel" name="pulse_label">
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Internal Pulse Modulation Settings</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="6" column="3">
       <widget class="QComboBox" name="sample_units_combobox">
        <property name="currentIndex">
         <number>-1</number>
        </property>
        <property name="placeholderText">
         <string>Units</string>
        </property>
       </widget>
      </item>
      <item row="11" column="3">
       <widget class="QPushButton" name="pulse_toggle_pushbutton">
        <property name="text">
         <string>Toggle Pulse</string>
        </property>
       </widget>
      </item>
      <item row="14" column="3">
       <widget class="QPushButton" name="dump_pulse_info_pushbutton">
        <property name="text">
         <string>Dump Pulse Info</string>
        </property>
       </widget>
      </item>
      <item row="11" column="1" colspan="2">
       <layout class="QHBoxLayout" name="pulse_settings_layout">
        <item>
         <widget class="QLineEdit" name="pulse_period_line">
          <property name="placeholderText">
           <string>Pulse Period</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="pulse_width_line">
          <property name="placeholderText">
           <string>Pulse Width</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="pulse_timeunits_combobox"/>
        </item>
       </layout>
      </item>
      <item row="9" column="3">
       <widget class="QPushButton" name="signal_setting_pushbutton">
        <property name="text">
         <string>Apply</string>
        </property>
       </widget>
      </item>
      <item row="9" column="0">
       <layout class="QHBoxLayout" name="signal_amplitude_layout">
        <item>
         <widget class="QLineEdit" name="signal_amplitude_line">
          <property name="placeholderText">
           <string>Amplitude</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="amplitude_units_combobox">
          <property name="placeholderText">
           <string>Units</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="8" column="0" colspan="4">
       <widget class="QLabel" name="signal_quickedit_label">
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Quick Settings</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QPushButton" name="try_ip_pushbutton">
        <property name="text">
         <string>Connect</string>
        </property>
       </widget>
      </item>
      <item row="7" column="3">
       <widget class="QPushButton" name="sample_frequency_pushbutton">
        <property name="text">
         <string>Sample</string>
        </property>
       </widget>
      </item>
      <item row="17" column="2">
       <widget class="QPushButton" name="external_start_pushbutton">
        <property name="text">
         <string>Start</string>
        </property>
       </widget>
      </item>
      <item row="17" column="3">
       <widget class="QPushButton" name="external_stop_pushbutton">
        <property name="text">
         <string>Stop</string>
        </property>
       </widget>
      </item>
      <item row="17" column="0" colspan="2">
       <widget class="QComboBox" name="external_modulation_combobox">
        <property name="placeholderText">
         <string>External Modulation Program</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MplWidget</class>
   <extends>QWidget</extends>
   <header>mplwidget</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <slots>
  <slot>connect_microwave_clicked()</slot>
 </slots>
</ui>
